### PR TITLE
:sparkles: Support obj to json.

### PR DESCRIPTION
**Pull request by iota9star**

### CODE REVIEW

File: lib/candid/idl.dart [hunk 1]
```diff
@@ -924,8 +966,11 @@ class RecordClass extends ConstructType<Map> {
   Uint8List encodeValue(Map x) {
     final values = _fields.map((entry) => x[entry.key]).toList();
 
-    final bufs = zipWith<MapEntry, dynamic, Uint8List>(
-        _fields, values, (entry, d) => entry.value.encodeValue(d));
+    final bufs =
+        zipWith<MapEntry, dynamic, Uint8List>(_fields, values, (entry, d) {
+      var t = entry.value;
+      return tryToJson(t, d) ?? t.encodeValue(d);
+    });
```
Comment (AlexV525) [inline lib/candid/idl.dart:973]:
```suggestion
    final bufs = zipWith<MapEntry, dynamic, Uint8List>(
      _fields,
      values,
      (entry, d) {
        var t = entry.value;
        return tryToJson(t, d) ?? t.encodeValue(d);
      },
    );
```

**Review by AlexV525 — APPROVED** [review]
LGTM with a nit

File: lib/candid/idl.dart (hunk 1 above)
Comment (neeboo) [reply lib/candid/idl.dart:973]: will write some test and confirm it

Comment (neeboo) [reply lib/candid/idl.dart:973]: LGTM, pass unit tests, merge it first

**Review by neeboo — APPROVED** [review]
LGTM

**Pull request merged**